### PR TITLE
DOMO-352334: Extracting expiration date from access token and checkin…

### PR DESCRIPTION
**Pull request by samrands**
…g if token is expired before making requests to improve efficiency and avoid sending a stream twice

### CODE REVIEW

**Comment by samrands** [comment]
Will there be any issues with importing base64 and datetime? Also, should I make any changes to the Changelog? I don't know the process for updating the SDK.

**Review by Gurfuzle — APPROVED** [review]
LTGM

File: pydomo/Transport.py [hunk 1]
```diff
@@ -71,23 +73,41 @@ def request(self, url, method, headers, params=None, body=None):
...
         response = requests.post(url=url, auth=HTTPBasicAuth(self.clientId, self.clientSecret))
         if response.status_code == requests.codes.OK:
             self.access_token = response.json()['access_token']
+            self.token_expiration = self._extract_expiration(self.access_token)
         else:
             self.logger.debug('Error retrieving access token: ' + self.dump_response(response))
             raise Exception("Error retrieving a Domo API Access Token: " + response.text)
 
+    def _extract_expiration(self, access_token):
+        EXPIRATION_DATE_KEY = 'exp'
```
Comment (thattolleyguy) [inline pydomo/Transport.py:97]: I would move this outside the method or get rid of it all together. It looks like we have keys commonly not set to a variable (see 'access_token' on line 90)

File: pydomo/Transport.py [hunk 2]
```diff
@@ -71,23 +73,41 @@ def request(self, url, method, headers, params=None, body=None):
...
+
+        token_parts = access_token.split('.')
+        payload_bytes = bytes(token_parts[1], 'utf-8')
```
Comment (thattolleyguy) [inline pydomo/Transport.py:100]: How do we handle if we have trouble parsing the token? Should we be catching all errors and returning unauthorized if there's a token error?

**Pull request merged**